### PR TITLE
[Companion] BEEFY crates renaming

### PR DESCRIPTION
**Pull request by davxy**
Companion for https://github.com/paritytech/substrate/pull/13484

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -452,63 +452,10 @@ dependencies = [
...
- "sp-core",
- "sp-runtime",
- "thiserror",
-]
-
 [[package]]
 name = "binary-merkle-tree"
 version = "4.0.0-dev"
-source = "git+https://github.com/paritytech/substrate?branch=master#7c8b89b3a6b603902709fc33fdb10a864b8b4204"
+source = "git+https://github.com/davxy/substrate?branch=move-beefy-to-consensus#10dffc0aba17b8037b6ca7ec49ca876037a1179d"
```
Comment (acatangiu) [inline Cargo.lock:458]: this should be git+https://github.com/paritytech/substrate

same across whole `Cargo.lock`

might be that the bot will automatically update these to the correct thing once the substrate PR is merged, but doing this is non-standard afaik

File: Cargo.toml [hunk 2]
```diff
@@ -32,7 +32,7 @@ assert_cmd = "2.0.4"
 nix = { version = "0.26.1", features = ["signal"] }
 tempfile = "3.2.0"
 tokio = "1.24.2"
-substrate-rpc-client = { git = "https://github.com/paritytech/substrate", branch = "master" }
+substrate-rpc-client = { git = "https://github.com/davxy/substrate", branch = "move-beefy-to-consensus" }
```
Comment (acatangiu) [inline Cargo.toml:35]: this should definitely not change

```suggestion
substrate-rpc-client = { git = "https://github.com/paritytech/substrate", branch = "master" }
```

Same for all `Cargo.toml` files

**Review by acatangiu — CHANGES_REQUESTED** [review]
You shouldn't change `Cargo.toml`s (other than renaming the beefy crates) - aka it should still point to `git = "https://github.com/paritytech/substrate", branch = "master"`

The companion system will do the rest of the magic:
1. it will `cargo patch` substrate to the right PR branch for the CI
2. once the PR is merged, it will update substrate references here to the right commit hash in `paritytech/substrate/master`

**Review by michalkucharczyk — APPROVED** [review]

File: node/service/Cargo.toml [hunk 3]
```diff
@@ -9,8 +9,7 @@ edition.workspace = true
 # Substrate Client
 sc-authority-discovery = { git = "https://github.com/paritytech/substrate", branch = "master" }
 babe = { package = "sc-consensus-babe", git = "https://github.com/paritytech/substrate", branch = "master" }
-beefy-primitives = { git = "https://github.com/paritytech/substrate", branch = "master", package = "sp-beefy" }
-beefy-gadget = { git = "https://github.com/paritytech/substrate", branch = "master" }
+beefy-gadget = { package = "sc-consensus-beefy", git = "https://github.com/paritytech/substrate", branch = "master" }
```
Comment (acatangiu) [inline node/service/Cargo.toml:12]: I would also change this (and the code references) to `sc-consensus-beefy` or `sc-beefy`;

Same for `beefy-primitives` -> `sp-beefy`

**Review by acatangiu — APPROVED** [review]

File: node/service/Cargo.toml [hunk 4]
```diff
@@ -35,6 +34,7 @@ telemetry = { package = "sc-telemetry", git = "https://github.com/paritytech/sub
 # Substrate Primitives
 sp-authority-discovery = { git = "https://github.com/paritytech/substrate", branch = "master" }
 consensus_common = { package = "sp-consensus", git = "https://github.com/paritytech/substrate", branch = "master" }
+beefy-primitives = { package = "sp-consensus-beefy", git = "https://github.com/paritytech/substrate", branch = "master" }
```
Comment (acatangiu) [inline node/service/Cargo.toml:37]:
```suggestion
sp-consensus-beefy = { package = "sp-consensus-beefy", git = "https://github.com/paritytech/substrate", branch = "master" }
```

Comment (davxy) [reply node/service/Cargo.toml:37]: Yeah. you are absolutely right! And I really would :-) 
But then if I follow my instinct I have to rename all this stuff as well:
- beefy-primitives -> sp-consensus-beefy
- grandpa-primitives -> sp-consensus-grandpa
- beefy -> sc-consensus-beefy
- grandpa -> sc-consensus-grandpa
- babe -> sc-consensus-babe
(for the coherence reason 😃)

So probably I would merge this first, just to let polkadot to compile with current substrate master. 
Then I can do that in a follow up PR

**Pull request merged**